lib/util/metadata_tools.rb

```diff
@@ -0,0 +1,187 @@
+module MetadataTools
+
+  def add_dataset_metadata(obj, orgs_cache)
+    d = Dataset.find_or_create_by(uuid: obj["id"])
+    d.name = obj["name"]
+    d.title = obj["title"]
+    d.summary = generate_summary(obj["notes"])
+    d.description = obj["notes"]
+    d.organisation_id = orgs_cache[obj["owner_org"]]
+    d.licence = obj["license_id"]
+    d.licence = 'no-licence' if d.licence == ""
+    d.licence_other = ""
+    #d.frequency = convert_frequency(obj["update_frequency"])
+    d.frequency = "never"
+    d.published = false
+    d.published_date = obj["metadata_created"]
+    d.created_at = obj["metadata_created"]
+    d.updated_at = obj["metadata_modified"]
+    d.dataset_type = dataset_type(obj)
+    d.harvested = harvested?(obj)
+    d.location1 = ""
+    d.location2 = ""
+    d.location3 = ""
+    d.legacy_metadata = ""
+    d.save!()
+
+    # Add the inspire metadata if we have determined this is a ULKP
+    # dataset.
+    if d.dataset_type == 'inspire'
+      inspire = add_inspire_metadata(d.id, obj)
+      inspire.save!()
+    end
+
+    # Iterate over the resources list and add a new datafile for each
+    # item.
+    obj["resources"].each do |resource|
+      add_resource(resource, d)
+    end
+
+    d.published = true
```

Comment:
i think it's assuming that everything in the import IS published, which may be true - @rossjones ?